open_humans/account_views.py

```diff
@@ -314,3 +312,29 @@ def dispatch(self, request, *args, **kwargs):
                 return complete_social_login(request, self.sociallogin)
 
         return ret
+
+
+class StoreRedirectURLView(View):
+    """
+    Endpoint for storing the redirect url in the session.
+    """
+
+    def post(self, request, *args, **kwargs):
+        default_redirect = reverse(settings.LOGIN_REDIRECT_URL)
+        raw_url = request.POST.get('next_url', default_redirect)
+        # Using window.location.href on the js side gives full domain + path,
+        # and we only want the path
+        parsed_url = urlparse(raw_url)
+        path = parsed_url.path
+        params = parsed_url.query
+        if params:
+            next_t = '{0}?{1}'.format(path, params)
+        else:
+            next_t = path
+        # In case someone tries to login from the signup page, it would
+        # otherwise inifinitely redirect, so we leave the session alone
+        if path != reverse('account_login'):
```

Comment:
maybe might as well do `if path not in [reverse('account_login'), reverse('account_signup')]'